src/parser/index.ts

```diff
@@ -18,46 +19,34 @@ export function parse(
   tokens: Array<T.Token>,
   parseOptions?: T.ParseOptions,
 ): T.AST {
+  const tt = new TokenTracker(tokens);
   const globalAst: T.AST = parseOptions?.ast ?? [];
   const globalScope: T.Scope = {
     parentScope: null,
     variables: parseOptions?.variables ?? (new Map<string, T.Variable>()),
     functions: parseOptions?.functions ?? (new Map<string, T.FunctionPattern>()),
   };
 
-  let index = 0;
-  let curTok = tokens[index];
-
-  function nextToken(): T.Token {
-    curTok = tokens[++index];
-    return curTok;
-  }
-
-  // TODO: Implement token tracker
-  function currentToken(): T.Token {
-    return curTok;
-  }
-
   function parseExpr(prevExpr?: T.Expr, meta?: any): T.Expr {
     const scope: T.Scope = meta?.scope ?? globalScope;
     const ast: T.AST = meta?.ast ?? globalAst;
 
-    if (curTok.type === 'keyword') {
-      if (curTok.value === 'def') {
-        return parseFunctionDeclaration(curTok, nextToken, parseExpr, scope);
+    if (tt.is('keyword')) {
+      if (tt.current.value === 'def') {
+        return parseFunctionDeclaration(tt, parseExpr, scope);
       }
 
-      if (curTok.value === 'if') {
+      if (tt.current.value === 'if') {
         let resultExpr: T.ConditionalExpr;
-        [curTok, resultExpr] = parseConditionalExpr(curTok, nextToken, parseExpr);
+        resultExpr = parseConditionalExpr(tt, parseExpr);
         return resultExpr;
       }
 
-      if (curTok.value === 'not') {
+      if (tt.current.value === 'not') {
         return parseLogicalNotExpr();
       }
 
-      if (curTok.value === 'and' || curTok.value === 'or') {
+      if (tt.current.value === 'and' || tt.current.value === 'or') {
         let resultExpr: T.Expr;
```

Comment:
Implement `tt.isOneOf` and `tt.isOneOfValue` to check whether token belongs to one of the types or values